src/main/java/com/pusher/client/channel/PrivateEncryptedChannelEventListener.java

```diff
@@ -0,0 +1,18 @@
+package com.pusher.client.channel;
+
+/**
+ * Interface to listen to private encrypted channel events.
+ */
+public interface PrivateEncryptedChannelEventListener extends ChannelEventListener {
+    /**
+     * Called when an attempt to authenticate a private encrypted channel fails.
+     *
+     * @param message
+     *            A description of the problem.
+     * @param e
+     *            An associated exception, if available.
+     */
+    void onAuthenticationFailure(String message, Exception e);
+
+    // todo: handle not receiving the shared secret?
```

Comment:
```suggestion
    //  TODO: add onDecryptionFailure
```
that to be more generic as we may get the key but it may be not valid (even after trying to the re-auth in such case), the exception and its message can serve to provide more specific details

Reply:
Yeah. That makes sense 👍